embassy-executor/src/raw/mod.rs

```diff
@@ -203,15 +208,22 @@ impl<F: Future + 'static> TaskStorage<F> {
     }
 
     unsafe fn poll(p: TaskRef) {
-        let this = &*(p.as_ptr() as *const TaskStorage<F>);
+        let this = &*p.as_ptr().cast::<TaskStorage<F>>();
 
         let future = Pin::new_unchecked(this.future.as_mut());
         let waker = waker::from_task(p);
         let mut cx = Context::from_waker(&waker);
         match future.poll(&mut cx) {
             Poll::Ready(_) => {
-                this.future.drop_in_place();
                 this.raw.state.despawn();
```

Comment:
the despawn call has to be last, because it's what "releases" the task storage.

Reply:
Yeah I noticed immediately, don't worry 😅